autoload/vimlparser.vim

```diff
@@ -3914,51 +3919,51 @@ function! s:Compiler.compile(node)
   if a:node.type == s:NODE_TOPLEVEL
     return self.compile_toplevel(a:node)
   elseif a:node.type == s:NODE_COMMENT
-    return self.compile_comment(a:node)
+    call self.compile_comment(a:node)
```

Comment:
Compiler.compile is only used for compiling Vim script AST into S-like-expression.

It's basically useful just for debugging vimlparser and not often called by other tools, so I think it's not performance sensitive code.

I added a commit which add "return s:NIL" for every non-return condition, but maybe we can revert the commit?